src/lib/Libtpp/tpp_router.c

```diff
@@ -316,11 +318,11 @@ broadcast_to_my_routers(tpp_chunk_t *chunks, int count, int origin_tfd)
 		}
 		if (tpp_enque(&router_list, r) == NULL) {
 			tpp_log(LOG_CRIT, __func__, "Out of memory enqueuing to router_list");
+			tpp_unlock_rwlock(&router_lock);
```

Comment:
where did we acquire this lock? it might be better to unlock in the same function that locks it, if possible.

Reply:
The lock is acquired in the calling function, please see the header